aws/components/apigateway/setup.ftl

```diff
@@ -303,27 +303,64 @@
+        [#local methodSettings = []]
+        [#local openapiIntegrations = 
+            getOccurrenceSettingValue(occurrence, [["apigw"], ["Integrations"]], true)]
+
+        [#if openapiIntegrations.Throttling?has_content]
+            [#local methodSettings = [
+                {
+                    "HttpMethod": "*",
+                    "ResourcePath": "/*",
+                    "LoggingLevel": "INFO",
+                    "DataTraceEnabled": true
+                } +
+                attributeIfContent(
+                    "ThrottlingBurstLimit", 
+                    openapiIntegrations.Throttling.BurstLimit) +
+                attributeIfContent(
+                    "ThrottlingRateLimit",
+                    openapiIntegrations.Throttling.RateLimit)
+            ]]
+        [/#if]
+
+        [#-- Integration Patterns (as Regex) into Matching Method Throttling (as explicit paths) --]
+        [#if definitionsObject[core.Id]?? ]
+            [#list definitionsObject[core.Id].paths as path,pathConfig]
+                [#list pathConfig?keys as verb]
+                    [#list openapiIntegrations.Patterns as pattern]
+                        [#if path?matches(pattern.Path)] 
+                            [#local methodSettings += 
+                            [
+                                {
+                                    "ResourcePath" : path,
+                                    "HttpMethod": pattern.Verb
+                                } +
+                                attributeIfContent("ThrottlingBurstLimit", pattern.BurstLimit) +
+                                attributeIfContent("ThrottlingRateLimit", pattern.RateLimit) 
+                            ]]
+                        [/#if]
+                    [/#list]
+                [/#list]
+            [/#list]
+        [/#if]
+
         [@cfResource
             id=stageId
             type="AWS::ApiGateway::Stage"
             properties=
                 {
                     "DeploymentId" : getReference(deployId),
                     "RestApiId" : getReference(apiId),
-                    "MethodSettings": [
-                        {
-                            "HttpMethod": "*",
-                            "ResourcePath": "/*",
-                            "LoggingLevel": "INFO",
-                            "DataTraceEnabled": true
-                        }
-                    ],
                     "StageName" : stageName,
                     "AccessLogSetting" : {
                         "DestinationArn" : getArn(accessLgId),
                         "Format" : "$context.identity.sourceIp $context.identity.caller $context.identity.user $context.identity.userArn [$context.requestTime] $context.apiId $context.httpMethod $context.resourcePath $context.protocol $context.status $context.responseLength $context.requestId"
                     }
                 } +
+                attributeIfContent("MethodSettings", methodSettings) +
```

Comment:
Again, my bad. We always want this attribute but I think we leave the code as you now have it. It works the same and would permit a case where we didn't want to always provide method defaults.